submittal/src/main/java/com/kyriba/submittal/service/TopicServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.kyriba.submittal.service;
+
+import com.kyriba.submittal.dao.TopicRepository;
+import com.kyriba.submittal.domain.Topic;
+import com.kyriba.submittal.domain.TopicStatus;
+import com.kyriba.submittal.service.exception.ObjectNotFoundException;
+import com.kyriba.submittal.service.exception.UnsupportedTopicTransitionException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * @author M-ABL
+ */
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class TopicServiceImpl implements TopicService
+{
+  private final TopicRepository topicRepository;
+
+
+  @Override
+  public void transit(final long topicId, @NonNull final TopicStatus status)
+  {
+    Topic topic = findTopicById(topicId);
+    if (topic.getStatus() == TopicStatus.DRAFT) {

```

Comment:
Imho, this logic should sit inside of Topic, the service should just find appropriate topic, delegate to it, and then publish notification.

Reply:
how to make a topic api in order to understand that status wasn't changed and the service must throw an exception?

Reply:
I'd suggest to use events, like it is done in examples in chapter 5 in the book "Microservices Patterns". But frankly, I'm not sure whether this case worth it